packages/schematics/angular/utility/project.ts

```diff
@@ -46,3 +71,19 @@ export function isWorkspaceSchema(workspace: any): workspace is WorkspaceSchema
 export function isWorkspaceProject(project: any): project is WorkspaceProject {
   return !!(project && (project as WorkspaceProject).projectType);
 }
+
+export function isProjectUsingIvy(host: Tree, project: WorkspaceProject): boolean {
+  const projectRoot = getProjectRoot(project);
+  const config = getProjectConfig(host, projectRoot + 'tsconfig.app.json');
```

Comment:
We need to read this from the workspace, we don't have any guarantee that this wan't changed.